### ISSUE TITLE
Realm crashes when I try to configure, to save, to update some staff.

### ISSUE THREAD

**Issue by alychkouski** [realm/realm-cocoa#3316]
## Goals
I'm trying to configure, to save, to update some staff. But always I receive crash.

https://gyazo.com/378c8a0ad91844066bd728856865a32f
## Expected Results

I expect that there are no crashes.

## Actual Results

https://gyazo.com/378c8a0ad91844066bd728856865a32f

## Code Sample

```
...
NSInteger const kDatabaseVersion = 1;
...
- (void)configureDatabase {
    RLMRealmConfiguration *config = [RLMRealmConfiguration defaultConfiguration];
    config.schemaVersion = kDatabaseVersion;
    config.migrationBlock = ^(RLMMigration *migration, uint64_t oldSchemaVersion) {

        if (oldSchemaVersion < kDatabaseVersion) {}
    };
    
    [RLMRealmConfiguration setDefaultConfiguration:config];
    
    [RLMRealm defaultRealm];
}
```

or I just try to add new item to Realm:

            @try {
                RLMRealm *realm = [RLMRealm defaultRealm];
                [realm beginWriteTransaction];
                [realm addObject:pair];
                [realm commitWriteTransaction];
            }

## Version of Realm and Tooling
I'm using Pods: Realm (0.98.4)

**Comment by jpsim** [comment]
Hi @alychkouski, I took the time to create a project with your sample code and haven't been unable to reproduce the issue you've reported: https://github.com/jpsim/GH3316

Could you please either let me know what my repro case is missing to trigger the problem you've reported, or better yet, send us a full Xcode project that triggers the issue?

Thanks!

**Comment by Cellane** [comment]
I think I'm having the same issue. I've added Realm to my project via CocoaPods with the `use_frameworks!` option on (might be important, probably?) and anytime I access the database, Realm would crash with `EXC_BAD_ACCESS (code=EXC_ARM_DA_ALIGN)`. Downgrading to 0.98.2 “solved” the issue for me. I will try to replicate the issue in an empty project this evening.

**Comment by Cellane** [comment]
Added pull request to the jpsim/GH3316 repository with changes to force the test project to crash.

**Comment by bdash** [comment]
This is the same crash as #3321 and was fixed in v0.98.5.
